src/connectiondataholder.cpp

```diff
@@ -224,14 +232,23 @@ void ConnectionDataHolder::updateIpAddress() {
   connect(request, &NetworkRequest::requestCompleted,
           [this](QNetworkReply*, const QByteArray& data) {
             logger.log() << "IP address request completed";
-
             QJsonDocument json = QJsonDocument::fromJson(data);
             if (json.isObject()) {
               QJsonObject obj = json.object();
 
+              QJsonValue country = obj.take("country");
+              if (country.toString().toLower() !=
+                  MozillaVPN::instance()->currentServer()->countryCode()) {
+                logger.log() << "Reported ip not in the right country, retry!";
+                m_updatingIpAddress = false;
+                updateIpAddress();
```

Comment:
if for any reason the server goes out of sync, this method generates a loop of updateIpAddress(). Do we care about checking the countryCode ?

Reply:
Sadly thats essential. Even after confirming we might encounter a stray request that did not yet use the vpn, for whatever reason. I would rather add an exit condition then removing this :)

Reply:
We spoke about this on slack. This code has to changed. Maybe a simple timer can help.